playbooks/aws/openshift-cluster/build_ami.yml

```diff
@@ -6,64 +6,44 @@
   - name: get the necessary vars for ami building
     include_vars: vars.yml
 
-  - name: create a vpc with the name <clusterid>
+  - name: create an instance and prepare for ami
     include_role:
-      name: openshift_aws_vpc
+      name: openshift_aws
+      tasks_from: build_ami.yml
     vars:
-      r_openshift_aws_vpc_clusterid: "{{ provision.clusterid }}"
-      r_openshift_aws_vpc_cidr: "{{ provision.vpc.cidr }}"
-      r_openshift_aws_vpc_subnets: "{{ provision.vpc.subnets }}"
-      r_openshift_aws_vpc_region: "{{ provision.region }}"
-      r_openshift_aws_vpc_tags: "{{ provision.vpc.tags }}"
-      r_openshift_aws_vpc_name: "{{ provision.vpc.name | default(provision.clusterid) }}"
-
-  - name: create aws ssh keypair
-    include_role:
-      name: openshift_aws_ssh_keys
-    vars:
-      r_openshift_aws_ssh_keys_users: "{{ provision.instance_users }}"
-      r_openshift_aws_ssh_keys_region: "{{ provision.region }}"
-
-  - name: fetch the default subnet id
-    ec2_vpc_subnet_facts:
+      openshift_aws_clusterid: "{{ provision.clusterid }}"
+      openshift_aws_region: "{{ provision.region }}"
+      openshift_aws_create_vpc: "{{ provision.create_vpc | bool }}"
+      openshift_aws_create_security_groups: "{{ provision.create_security_groups | bool }}"
+      openshift_aws_create_ssh_keys: "{{ provision.create_ssh_keys | bool }}"
+      openshift_aws_users: "{{ provision.instance_users }}"
+      openshift_aws_node_group_type: compute
+      openshift_aws_vpc_subnet_name: us-east-1c
+      openshift_aws_ssh_key_name: "{{ provision.ssh_key_name }}"
+      openshift_aws_base_ami: "{{ provision.build.base_image }}"
+
+  - name: fetch newly created instances
+    ec2_remote_facts:
       region: "{{ provision.region }}"
       filters:
-        "tag:Name": "{{ provision.vpc.subnets[provision.region][0].az }}"
-    register: subnetout
-
-  - name: create instance for ami creation
-    ec2:
-      assign_public_ip: yes
-      region: "{{ provision.region }}"
-      key_name: "{{ provision.node_group_config.ssh_key_name }}"
-      group: "{{ provision.clusterid }}"
-      instance_type: m4.xlarge
-      vpc_subnet_id: "{{ subnetout.subnets[0].id }}"
-      image: "{{ provision.build.base_image }}"
-      volumes:
-      - device_name: /dev/sdb
-        volume_type: gp2
-        volume_size: 100
-        delete_on_termination: true
-      wait: yes
-      exact_count: 1
-      count_tag:
-        Name: ami_base
-      instance_tags:
-        Name: ami_base
-    register: amibase
+        "tag:Name": ami_base
```

Comment:
This needs to be configurable.

Reply:
This is a temporary build instance.  The name just needs to be unique.  I can make it configurable and default to `ami_base`  or `openshift_ami_base`.

Reply:
I actually need this to be configurable to deploy in our environment on aws.  Our instance name tags need to be specific or they might get killed.